packages/kbn-expandable-flyout/src/components/settings_menu.test.tsx

```diff
@@ -42,48 +55,131 @@ describe('SettingsMenu', () => {
     expect(getByTestId(SETTINGS_MENU_FLYOUT_TYPE_BUTTON_GROUP_TEST_ID)).toBeInTheDocument();
   });
 
+  it('should have the type selected if option is enabled', () => {
+    const state = {
+      panels: initialPanelsState,
+      ui: {
+        pushVsOverlay: 'push' as const,
+      },
+    };
+    const flyoutCustomProps = {
+      hideSettings: false,
+      pushVsOverlay: {
+        disabled: false,
+        tooltip: '',
+      },
+    };
+
+    const { getByTestId } = render(
+      <TestProvider state={state}>
+        <SettingsMenu flyoutCustomProps={flyoutCustomProps} />
+      </TestProvider>
+    );
+
+    getByTestId(SETTINGS_MENU_BUTTON_TEST_ID).click();
+
+    expect(getByTestId(SETTINGS_MENU_FLYOUT_TYPE_BUTTON_GROUP_PUSH_TEST_ID)).toHaveClass(
+      'euiButtonGroupButton-isSelected'
+    );
+    expect(getByTestId(SETTINGS_MENU_FLYOUT_TYPE_BUTTON_GROUP_OVERLAY_TEST_ID)).not.toHaveClass(
+      'euiButtonGroupButton-isSelected'
+    );
+  });
+
   it('should select correct the flyout type', () => {
-    const onChange = jest.fn();
-    const flyoutTypeProps = {
-      type: 'overlay' as EuiFlyoutProps['type'],
-      onChange,
-      disabled: false,
-      tooltip: '',
+    const flyoutCustomProps = {
+      hideSettings: false,
+      pushVsOverlay: {
+        disabled: false,
+        tooltip: '',
+      },
     };
 
-    const { getByTestId } = render(<SettingsMenu flyoutTypeProps={flyoutTypeProps} />);
+    const { getByTestId } = render(
+      <TestProvider>
+        <SettingsMenu flyoutCustomProps={flyoutCustomProps} />
+      </TestProvider>
+    );
+
+    expect(localStorage.getItem(EXPANDABLE_FLYOUT_LOCAL_STORAGE)).toEqual(null);
 
     getByTestId(SETTINGS_MENU_BUTTON_TEST_ID).click();
     getByTestId(SETTINGS_MENU_FLYOUT_TYPE_BUTTON_GROUP_PUSH_TEST_ID).click();
 
-    expect(onChange).toHaveBeenCalledWith('push');
+    expect(localStorage.getItem(EXPANDABLE_FLYOUT_LOCAL_STORAGE)).toEqual(
+      JSON.stringify({ [PUSH_VS_OVERLAY_LOCAL_STORAGE]: 'push' })
+    );
+
+    getByTestId(SETTINGS_MENU_FLYOUT_TYPE_BUTTON_GROUP_OVERLAY_TEST_ID).click();
+
+    expect(localStorage.getItem(EXPANDABLE_FLYOUT_LOCAL_STORAGE)).toEqual(
+      JSON.stringify({ [PUSH_VS_OVERLAY_LOCAL_STORAGE]: 'overlay' })
+    );
+  });
+
+  it('should not save to local storage if the option is disabled', () => {
```

Comment:
I think that test was actually overkill as we can't really click on the button if it's disabled anyway... but I've merged the code as suggested as it can't hurt to over test!